app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt

```diff
@@ -101,23 +99,24 @@ class RoutePresenter(
             is RouteAction.Dialog.SecurityDisclaimer -> showSecurityDisclaimerDialog()
             is RouteAction.Dialog.UnlinkDisclaimer -> showUnlinkDisclaimerDialog()
             is RouteAction.Dialog.NoNetworkDisclaimer -> showNoNetworkDialog()
+            is RouteAction.Dialog.OnboardingSecurityDialog -> showOnboardingSecurityDialog()
         }
 
         dialogStateObservable
-            .map { alertState ->
-                val action: Action? = when (alertState) {
+            .subscribe { alertState ->
```

Comment:
bonus points: to keep the the minimal `dispatcher::dispatch` subscription here, you could refactor this to:
```kotlin
.map { 
    // alertState -> List<Action> 
}
.flatMapIterable { it }
.subscribe(dispatcher::dispatch)
```

Reply:
reference: https://github.com/ReactiveX/RxJava/issues/2473